src/ir/properties.h

```diff
@@ -514,6 +514,38 @@ inline MemoryOrder getMemoryOrder(Expression* curr) {
   return MemoryOrder::Unordered;
 }
 
+inline bool hasUnwritableTypeImmediate(Expression* curr) {
+#define DELEGATE_ID curr->_id
+
+#define DELEGATE_FIELD_REF_CHILD(id, field)                                    \
+  {                                                                            \
+    auto type = curr->cast<id>()->field->type;                                 \
+    if (type == Type::unreachable || type.isNull()) {                          \
```

Comment:
Please add a comment on the function, mentioning the issue is that this is for the binary format, and why either unreachable or null cause issues there.